src/highdicom/sr/templates.py

```diff
@@ -1955,19 +3248,454 @@ def __init__(
+            except IndexError:
+                next_index = -1
+            if item.value == codes.DCM.Device:
+                attributes = DeviceObserverIdentifyingAttributes.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Person:
+                attributes = PersonObserverIdentifyingAttributes.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected observer type "{item.meaning}".')
+            context = ObserverContext(
+                observer_type=item.value,
+                observer_identifying_attributes=attributes
+            )
+            observer_contexts.append(context)
+        return observer_contexts
+
+    def get_subject_contexts(
+        self,
+        subject_class: Optional[CodedConcept] = None
+    ) -> List[SubjectContext]:
+        """Get subject contexts.
+
+        Parameters
+        ----------
+        subject_class: Union[highdicom.sr.coding.CodedConcept, pydicom.sr.coding.Code], optional
+            Type of subject ("Specimen", "Fetus", or "Device") for which should
+            be filtered
+
+        Returns
+        -------
+        List[highdicom.sr.SubjectContext]
+           Subject contexts
+
+        """  # noqa: E501
+        root_item = self[0]
+        matches = [
+            (i + 1, item) for i, item in enumerate(root_item.ContentSequence)
+            if item.name == codes.DCM.SubjectClass
+        ]
+        subject_contexts = []
+        for i, (index, item) in enumerate(matches):
+            if subject_class is not None:
+                if item.value != subject_class:
+                    continue
+            try:
+                next_index = matches[i + 1][0]
+            except IndexError:
+                next_index = -1
+            if item.value == codes.DCM.Specimen:
+                attributes = SubjectContextSpecimen.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Fetus:
+                attributes = SubjectContextFetus.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Device:
+                attributes = SubjectContextDevice(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected subject class "{item.meaning}".')
+            context = SubjectContext(
+                subject_class=item.value,
+                subject_class_specific_context=attributes
+            )
+            subject_contexts.append(context)
+        return subject_contexts
+
+    def get_planar_roi_measurement_groups(
+        self,
+        tracking_uid: Optional[str] = None,
```

Comment:
Add tracking_identifier filter in addition to tracking UID?

Reply:
At this point, I am reluctant to add more filters. Unless you really need this, I would (at least for now) focus on getting the filters right that we already implemented.

Reply:
I think this could be very commonly used for ML projects, even if using the UID would arguably be better. The reason to add it now would be that if we wait until later, it will have to go to the end of the function signature for backwards compatibility, when it probably belongs next to tracking_uid